File: PracticeAndroid/AndroidPractice2/MyFolder/LOCOMO/app/src/main/java/com/example/locomo/fragment/FirstFragment.java
package com.example.locomo.fragment;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.example.locomo.R;
import com.example.locomo.activity.DashBoardActivity;

public class FirstFragment extends Fragment implements View.OnClickListener {
    Button checkout_btn,ADExpress_btn;
    public FirstFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view= inflater.inflate(R.layout.fragment_first, container, false);
        checkout_btn=view.findViewById(R.id.checkout_btn);
        ADExpress_btn=view.findViewById(R.id.AdExpress_btn);
        checkout_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                FragmentTransaction fr = getParentFragmentManager().beginTransaction();
                fr.replace(R.id.main_frame,new SecondFragment());
                fr.commit();
            }
        });
        return view;
    }

    @Override
    public void onClick(View v) {

    }
}

File: PracticeAndroid/Andorid_Kotlin/ADExpress/Camerax_MVVM/app/src/main/java/com/androdevdk/camerax_mvvm/test/fragments/FirstFragment.kt
package com.androdevdk.camerax_mvvm.test.fragments

import android.app.Activity
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.androdevdk.camerax_mvvm.ApplicationClass
import com.androdevdk.camerax_mvvm.R
import com.androdevdk.camerax_mvvm.databinding.FragmentCameraBinding
import com.androdevdk.camerax_mvvm.databinding.FragmentFirstBinding
import com.androdevdk.camerax_mvvm.test.TestActivity
import com.androdevdk.camerax_mvvm.test.adapter.PersonAdapter


class FirstFragment : Fragment() {
    private var _fragmentFirstBinding: FragmentFirstBinding? = null
    private val fragmentFirstBinding get() = _fragmentFirstBinding!!
    var adapter: PersonAdapter? = null
    lateinit var applicationClass: ApplicationClass
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        _fragmentFirstBinding = FragmentFirstBinding.inflate(inflater, container, false)
        return fragmentFirstBinding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        applicationClass = ApplicationClass.instance
        fragmentFirstBinding.recycler.layoutManager = LinearLayoutManager(requireContext())
        adapter = PersonAdapter(
            requireContext(),
            applicationClass.data
        )
        fragmentFirstBinding.recycler.adapter = adapter
    }

    companion object {

    }
}